File: src/main/scala/kim/dencka/easy/collection/PalindromeNumber.scala
package kim.dencka.easy.collection

object PalindromeNumber extends App {

  def isPalindrome(x: Int): Boolean = {
    @annotation.tailrec
    def run(n: Int, digits: List[Int]): List[Int] =
      if (n < 10) n :: digits
      else run(n / 10, (n % 10) :: digits)

    val list = run(x, List.empty)

    if (x < 0) false
    else list == list.reverse
  }

  println(isPalindrome(100))
  println(isPalindrome(101))
  println(isPalindrome(-101))
  println(isPalindrome(2110112))
  println(isPalindrome(1000011))

}

File: src/main/scala/kim/dencka/java/easy/other/PalindromeNumber.java
package kim.dencka.java.easy.other;

class PalindromeNumber {
    public static void main(String[] args) {
        System.out.println(isPalindrome(-121));
    }

    public static boolean isPalindrome(int x) {
        if (x < 0) return false;
        int reversed = 0;
        int temp = x;
        while (temp != 0) {
            reversed = reversed  * 10 + (temp % 10);
            temp /= 10;
        }
        return reversed == x;
    }
}
